pkg/api/norman/customization/aks/handler.go

```diff
@@ -154,6 +165,33 @@ func (h *handler) getCloudCredential(req *http.Request, cap *Capabilities, credI
 	cap.ClientID = string(cc.Data["azurecredentialConfig-clientId"])
 	cap.ClientSecret = string(cc.Data["azurecredentialConfig-clientSecret"])
 
+	if cap.TenantID == "" {
+		annotations := cc.GetAnnotations()
+		tenantAnno, timestamp := annotations[tenantIDAnnotation], annotations[tenantIDTimestampAnnotation]
+		if tenantAnno != "" && timestamp != "" {
+			parsedTime, err := time.Parse(time.RFC3339, timestamp)
+			if err != nil {
+				return httperror.ServerError.Status, err
+			}
+			if parsedTime.Add(tenantIDTimeout).After(time.Now().UTC()) {
+				cap.TenantID = tenantAnno
+			}
+		}
```

Comment:
The timestamp is the whole point, we want to refresh it in case it has changed. If we weren't refreshing it we might as well store it once in the secret data from a controller.